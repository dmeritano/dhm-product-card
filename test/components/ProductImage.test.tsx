import * as React from 'react';
import renderer from "react-test-renderer"
import { ProductImage, ProductCard } from "../../src/components"
import { product2 } from '../data/products';


describe("ProductImage", () => {
    test("Should show the component with custom image", () => {

        const wrapper = renderer.create(
            <ProductImage img="../data/my-product-custom-image.png"/>
        )

        expect ( wrapper.toJSON()).toMatchSnapshot()
    })

    
    test("Should show the component with product image", () => {

        const wrapper = renderer.create(
            
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage/>
                    )
                }
            </ProductCard>
            
        )
        expect ( wrapper.toJSON()).toMatchSnapshot()
    })
    

})