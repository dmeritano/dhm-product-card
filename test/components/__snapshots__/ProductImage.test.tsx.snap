// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Should show the component with custom image 1`] = `
<img
  alt="Product img"
  className="productImg undefined"
  src="../data/my-product-custom-image.png"
/>
`;

exports[`ProductImage Should show the component with product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product img"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
